contrib/openbsm/bin/bsmconv/bsmconv.c

```diff
@@ -0,0 +1,101 @@
+#include <sys/time.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <bsm/libbsm.h>
+#include <bsm/audit_record.h>
+
+#include "linau.h"
+#include "pjdlog.h"
+
+#define BSMCONV_BUFFER_SIZE 8192
+
+
+static void	process_events(FILE *fp);
+static void	process_event(const struct linau_event *event);
+static void	parse_command_line_options(int argc, char **argv,
+		    int *debuglevelp);
+
+
+static void
+process_event(const struct linau_event *event)
+{
+	size_t buflen;
+	u_char buf[BSMCONV_BUFFER_SIZE];
+	int aurecordd;
+	struct timeval *tm;
+	unsigned short aueventid;
+
+	PJDLOG_ASSERT(event != NULL);
+
+	aurecordd = linau_event_to_au(event, &aueventid);
+	tm = linau_event_get_timeval(event);
+
+	buflen = BSMCONV_BUFFER_SIZE;
+	PJDLOG_VERIFY(
+	    au_close_buffer_tm(aurecordd, aueventid, buf, &buflen, tm) == 0);
+
+	write(1, buf, buflen);
+
+	free(tm);
+}
+
+static void
+process_events(FILE *fp)
+{
+	struct linau_event *event;
+	struct linau_record *record;
+
+	PJDLOG_ASSERT(fp != NULL);
+
+	event = linau_event_create();
+	PJDLOG_VERIFY(event != NULL);
```

Comment:
PJDLOG_VERIFY should be used to verify a return value of a function without an assignment. Use PJDLOG_ASSERT here.